crash with unknown cause - background run

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Unknown - background run

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
N/a

**Version Info:**
 - App Version: 0.17.3
 - Android Version: 13
 - Device Model: galaxy a51

**Additional context**
Add any other information about the problem here.
Suntimes 0.17.3 (133) [9fae5276d]
on Android 13 (33) [samsung]

java.lang.RuntimeException: Unable to start receiver com.forrestguice.suntimeswidget.widgets.SuntimesWidget0: java.lang.IllegalArgumentException: Total seconds out of range while fraction is non-zero: 41958
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4315)
	at android.app.ActivityThread.-$$Nest$mhandleReceiver(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.IllegalArgumentException: Total seconds out of range while fraction is non-zero: 41958
	at net.time4j.tz.ZonalOffset.<init>(ZonalOffset.java:114)
	at net.time4j.tz.ZonalOffset.atLongitude(ZonalOffset.java:224)
	at net.time4j.calendar.astro.StdSolarCalculator$3.event(StdSolarCalculator.java:559)
	at net.time4j.calendar.astro.StdSolarCalculator$3.sunrise(StdSolarCalculator.java:539)
	at net.time4j.calendar.astro.SolarTime$1.apply(SolarTime.java:615)
	at net.time4j.calendar.astro.SolarTime$1.apply(SolarTime.java:612)
	at net.time4j.engine.ChronoEntity.get(ChronoEntity.java:160)
	at com.forrestguice.suntimeswidget.calculator.time4a.Time4ASuntimesCalculator.getOfficialSunriseCalendarForDate(Time4ASuntimesCalculator.java:137)
	at com.forrestguice.suntimeswidget.calculator.SuntimesRiseSetData.calculate(SuntimesRiseSetData.java:506)
	at com.forrestguice.suntimeswidget.widgets.SuntimesWidget0.createRiseSetData(SuntimesWidget0.java:742)
	at com.forrestguice.suntimeswidget.widgets.SuntimesWidget0.updateAppWidget(SuntimesWidget0.java:705)
	at com.forrestguice.suntimeswidget.widgets.SuntimesWidget0.updateAppWidget(SuntimesWidget0.java:624)
	at com.forrestguice.suntimeswidget.widgets.SuntimesWidget0.updateWidget(SuntimesWidget0.java:424)
	at com.forrestguice.suntimeswidget.widgets.SuntimesWidget0.onUpdate(SuntimesWidget0.java:414)
	at android.appwidget.AppWidgetProvider.onReceive(AppWidgetProvider.java:71)
	at com.forrestguice.suntimeswidget.widgets.SuntimesWidget0.onReceive(SuntimesWidget0.java:155)
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:4306)
	... 9 more

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,8 @@
 ### ~
 
 ### v0.17.4 (2026-07-26)
+* fixes app crash when using `cc` calculator with longitudes over +-165 degrees (#966).
+* fixes app crash when changing widget padding (#963).
 * fixes bug where "latest sunset time is slightly off" (#960).
 * fixes bug where "sunlight dialog does not honor the `show seconds` option".
 * fixes bug where "sunlight dialog `share` does not honor the `time zone` option".

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 * fixes bug where "latest sunset time is slightly off" (#960).
 * fixes bug where "sunlight dialog does not honor the `show seconds` option".
 * fixes bug where "sunlight dialog `share` does not honor the `time zone` option".
+* build: updated to support jdk21 (minimum required is now jdk17+); updates gradle `7.4.0 -> 8.5.2`; updates agp `7.5 -> 8.7`.
 * updates translation to Italian (it) (#964 by McCio).
 * updates translations to Simplified Chinese (zh_CN) and Traditional Chinese (zh_TW) (#962 by James Liu).
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,12 @@
 ### ~
 
+### v0.17.4 (2026-06-26)
+* fixes bug where "latest sunset time is slightly off" (#960).
+* fixes bug where "sunlight dialog does not honor the `show seconds` option".
+* fixes bug where "sunlight dialog `share` does not honor the `time zone` option".
+* updates translation to Italian (it) (#964 by McCio).
+* updates translations to Simplified Chinese (zh_CN) and Traditional Chinese (zh_TW) (#962 by James Liu).
+
 ### v0.17.3 (2026-06-22)
 * extends the "set background" world map action; additional map backgrounds are now available as an addon (https://github.com/forrestguice/SuntimesMapPack).
 * fixes inconsistent Mercator and Van der Grinten projection extents.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 ### ~
 
-### v0.17.4 (2026-06-26)
+### v0.17.4 (2026-07-26)
 * fixes bug where "latest sunset time is slightly off" (#960).
 * fixes bug where "sunlight dialog does not honor the `show seconds` option".
 * fixes bug where "sunlight dialog `share` does not honor the `time zone` option".

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -81,8 +81,8 @@ android
             missingDimensionStrategy 'api', 'api28'
         }
 
-        versionCode 133
-        versionName "0.17.3"
+        versionCode 134
+        versionName "0.17.4"
 
         manifestPlaceholders = [suntimesAuthorityRoot:"suntimes",
                                 suntimesAuthorityRoot0:"suntimeswidget",

--- a/fastlane/metadata/android/en-US/changelogs/134.txt
+++ b/fastlane/metadata/android/en-US/changelogs/134.txt
@@ -1,0 +1,3 @@
+- fixes bug where "latest sunset time is slightly off" (#960).
+- updates translation to Italian.
+- updates translations to Simplified Chinese and Traditional Chinese.

--- a/fastlane/metadata/android/en-US/changelogs/134.txt
+++ b/fastlane/metadata/android/en-US/changelogs/134.txt
@@ -1,3 +1,5 @@
+- fixes app crash when using `cc` calculator with longitudes over +-165 degrees (#966).
+- fixes app crash when changing widget padding (#963).
 - fixes bug where "latest sunset time is slightly off" (#960).
 - updates translation to Italian.
 - updates translations to Simplified Chinese and Traditional Chinese.

